postgresql/resource_postgresql_grant.go

```diff
@@ -135,14 +136,26 @@ func resourcePostgreSQLGrantCreate(d *schema.ResourceData, meta interface{}) err
 	}
 	defer deferredRollback(txn)
 
-	// Revoke all privileges before granting otherwise reducing privileges will not work.
-	// We just have to revoke them in the same transaction so the role will not lost its
-	// privileges between the revoke and grant statements.
-	if err = revokeRolePrivileges(txn, d); err != nil {
+	owners, err := getRolesToGrantForSchema(txn, schemaName)
+	if err != nil {
 		return err
 	}
 
-	if err = grantRolePrivileges(txn, d); err != nil {
+	if err = withRolesGranted(txn, owners, func() error {
+
+		// Revoke all privileges before granting otherwise reducing privileges will not work.
+		// We just have to revoke them in the same transaction so the role will not lost its
+		// privileges between the revoke and grant statements.
+		if err = revokeRolePrivileges(txn, d); err != nil {
+			return err
+		}
+
+		if err = grantRolePrivileges(txn, d); err != nil {
+			return err
+		}
+
+		return nil
```

Comment:
```suggestion
	if err := withRolesGranted(txn, owners, func() error {
		// Revoke all privileges before granting otherwise reducing privileges will not work.
		// We just have to revoke them in the same transaction so the role will not lost its
		// privileges between the revoke and grant statements.
		if err := revokeRolePrivileges(txn, d); err != nil {
			return err
		}
		if err := grantRolePrivileges(txn, d); err != nil {
			return err
		}
		return nil
```